src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs

```diff
@@ -177,28 +183,79 @@ public static DiagnosticData CreateAnalyzerLoadFailureDiagnostic(string fullPath
+                    description = CreateDescription(e.Exception);
                     break;
                 case AnalyzerLoadFailureEventArgs.FailureErrorCode.UnableToCreateAnalyzer:
                     id = Choose(language, WRN_AnalyzerCannotBeCreatedId, WRN_AnalyzerCannotBeCreatedIdCS, WRN_AnalyzerCannotBeCreatedIdVB);
                     messageFormat = FeaturesResources.WRN_AnalyzerCannotBeCreated;
                     message = string.Format(FeaturesResources.WRN_AnalyzerCannotBeCreated, e.TypeName, fullPath, e.Message);
+                    description = CreateDescription(e.Exception);
                     break;
                 case AnalyzerLoadFailureEventArgs.FailureErrorCode.NoAnalyzers:
                     id = Choose(language, WRN_NoAnalyzerInAssemblyId, WRN_NoAnalyzerInAssemblyIdCS, WRN_NoAnalyzerInAssemblyIdVB);
                     messageFormat = FeaturesResources.WRN_NoAnalyzerInAssembly;
                     message = string.Format(FeaturesResources.WRN_NoAnalyzerInAssembly, fullPath);
+                    description = CreateDescription(e.Exception);
                     break;
                 case AnalyzerLoadFailureEventArgs.FailureErrorCode.None:
                 default:
                     id = string.Empty;
                     message = string.Empty;
                     messageFormat = string.Empty;
+                    description = string.Empty;
                     return false;
             }
 
             return true;
         }
 
+        private static string CreateDescription(Exception exception)
+        {
+            var aggregateException = exception as AggregateException;
+            if (aggregateException != null)
+            {
+                var flattened = aggregateException.Flatten();
+                return string.Join(Separator, flattened.InnerExceptions.Select(e => GetExceptionMessage(e)));
+            }
+
+            if (exception != null)
+            {
+                return string.Join(Separator, GetExceptionMessage(exception), CreateDescription(exception.InnerException));
+            }
+
+            return string.Empty;
+        }
+
+        private static string GetExceptionMessage(Exception exception)
+        {
+            var fileNotFoundException = exception as FileNotFoundException;
+            if (fileNotFoundException == null)
+            {
+                return exception.Message;
+            }
+
+            var fusionLog = GetFusionLogIfPossible(fileNotFoundException);
+            if (fusionLog == null)
+            {
+                return exception.Message;
+            }
+
+            return string.Join(Separator, fileNotFoundException.Message, fusionLog);
+        }
+
+        private static string GetFusionLogIfPossible(FileNotFoundException exception)
+        {
+            try
+            {
+                var info = exception.GetType().GetRuntimeProperty("FusionLog");
```

Comment:
I know some might not like this but FusionLog gives most detail info on why it is failed on Desktop. and since Feature is portable layer, getting this info in other way is way more complex.